### PR TITLE
add simple script-based reporter

### PR DESCRIPTION
**Pull request by kevinkreiser**
this python program mimics everything that is happening between the java kafka workers and the python reporter service but doesn't require kafka or running and http server. instead its meant to be run on the command line to import a chunk of data is if it were the java kafka worker PUTing files to s3.

i'm still in the testing phases but all of the parts are there and they seemed to work in isolation so after working out a few kinks of chaining them together we should be able to give this a shot for a simpler way to load historical dumps of data.

### CODE REVIEW

File: py/reporter_service.py [hunk 1]
```diff
@@ -79,6 +79,112 @@ def handle_request(self):
 class ThreadedHTTPServer(ThreadPoolMixIn, HTTPServer):
   pass
 
+#report some segments for the datastore
+def report(segments, trace, threshold_sec, report_levels, transition_levels):
```
Comment (kevinkreiser) [inline py/reporter_service.py:83]: first thing to note is that to make this easy on myself and not change the logic of the actual reporting of segments is that i broke this function out so i could import the reporter service as a module and call this one important function

File: src/main/java/io/opentraffic/reporter/Segment.java [hunk 2]
```diff
@@ -16,33 +16,18 @@
   public static long INVALID_SEGMENT_ID = 0x3fffffffffffL;
   public long id;         //main segment id
   public long next_id;    //optional next, could be invalid
-  public long min, max;   //epoch seconds
```
Comment (kevinkreiser) [inline src/main/java/io/opentraffic/reporter/Segment.java:19]: when porting some of this logic to python i noticed that there was still vestigial code in there that was both wasting space and made for reading of the code a bit confusing so i cleaned it up for my understanding and others

File: py/simple_reporter.py [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+  parser.add_argument('--src-time-pattern', type=str, help='A string used to extract epoch seconds from a time string', default='%Y-%m-%d %H:%M:%S')
+  parser.add_argument('--match-config', type=str, help='A file containing the config for the map matcher', required=True)
+  parser.add_argument('--quantisation', type=int, help='How large are the buckets to make tiles for. They should always be an hour (3600 seconds)', default=3600)
+  parser.add_argument('--privacy', type=int, help='How many readings of a given segment pair must appear before it being reported', default=2)
+  parser.add_argument('--source-id', type=str, help='A simple string to identify where these readings came from', default='smpl_rprt')
+  parser.add_argument('--dest-bucket', type=str, help='Bucket where we want to put the reporter output', required=True)
+  parser.add_argument('--concurrency', type=int, help='Number of threads to use when doing various stages of processing', default=multiprocessing.cpu_count())
+  args = parser.parse_args()
+
```
Comment (kevinkreiser) [inline py/simple_reporter.py]: 3 main steps:

* spawn threads to download source trace data from s3, each thread will gunzip the file and parse out individual lines and forward them to a file for just that vehicle for that line. the final result is one file for every uniq vehicle (we'll need a decent size inode table for large datasets)
* spawn threads to take each file of per vehicle trace, do the match to the osmlr segments and then store the result to time tiles (what the java reporter outputs)
* spawn threads to take each time tile and remove the observations that dont meet the privacy parameter, then upload that file to s3

details of each can be found above

**Comment by dgearhart** [comment]
I am assuming your are running a bbox and then we can review

**Comment by kevinkreiser** [comment]
yes ive added a bbox locally and a few typos and some error handling. when all the tests check out ill let you know!

**Comment by heffergm** [comment]
:ship: :it: 👍

**Pull request merged**